packages/@ama-sdk/schematics/schematics/typescript/core/index.ts

```diff
@@ -166,24 +168,50 @@ const getGeneratorOptions = (tree: Tree, context: SchematicContext, options: NgG
  */
 function ngGenerateTypescriptSDKFn(options: NgGenerateTypescriptSDKCoreSchematicsSchema): Rule {
 
-  return (tree, context) => {
+  return async (tree, context) => {
     const targetPath = options.directory || '';
     const generatorOptions = getGeneratorOptions(tree, context, options);
+    let isJson = false;
+    let specDefaultPath = path.posix.join(targetPath, `${LOCAL_SPEC_FILENAME}.json`);
+    specDefaultPath = existsSync(specDefaultPath) ? specDefaultPath : path.posix.join(targetPath, `${LOCAL_SPEC_FILENAME}.yaml`);
+    generatorOptions.specPath ||= specDefaultPath;
+
+    let specContent!: string;
+    if (URL.canParse(generatorOptions.specPath) && (new URL(generatorOptions.specPath)).protocol.startsWith('http')) {
```

Comment:
Should we allow download via http (instead of tls) ? Is it a practice we want to support?

Reply:
It is based on an input. It is on the user responsibility to provide a trusted link.
I don't see reason to restrain the user on that (especially if the user is in a secure network in which he access to the spec via http)